packages/p2p/src/components/app-content.jsx

```diff
@@ -13,18 +13,36 @@ import { useModalManagerContext } from 'Components/modal-manager/modal-manager-c
 import TemporarilyBarredHint from 'Components/temporarily-barred-hint';
 import { buy_sell } from 'Constants/buy-sell';
 import { useStores } from 'Stores';
-
+import { getHoursDifference } from 'Utils/date-time';
 import { localize } from './i18next';
 
 const AppContent = ({ order_id }) => {
     const { buy_sell_store, general_store } = useStores();
     const { showModal, hideModal } = useModalManagerContext();
+    let interval;
     const {
         notifications: { setP2POrderProps },
+        client: { loginid },
     } = useStore();
     const notification_count = useP2PNotificationCount();
     const history = useHistory();
 
+    // Check if it has been 24 hours since the user last saw the disclaimer modal
+    const checkShowDisclaimerModal = () => {
+        if (!(getHoursDifference(localStorage.getItem(`${loginid}_disclaimer_shown`)) < 24)) {
+            showModal({ key: 'DisclaimerModal' });
+        }
+    };
+
+    React.useEffect(() => {
+        checkShowDisclaimerModal();
+        interval = setInterval(() => {
+            checkShowDisclaimerModal();
+            // checks whether to show the disclaimer modal every hour
+        }, 3600000);
```

Comment:
why do we have this interval?

Reply:
runs every hour to check if 24 hours is complete

Reply:
doesnt this mean every time we reload the page, the interval will reset back to 24 hours, better to check on load the localStorage disclaimer item and get the hours from there which can be then used as the time for the interval

Reply:
no.. it doesn't reset to 24 hours.. the check for 24 hours is done based on the local storage value.

Reply:
ah yeah, my bad yes it runs every hour 👀 but why check every hour, would it not suffice if we check the return value of getHoursDifference and set the timer using that value?